src/View/RightPanel.coffee

```diff
@@ -0,0 +1,44 @@
+R = require "./R"
+Model = require "../Model/Model"
+Util = require "../Util/Util"
+
+
+R.create "RightPanel",
+  contextTypes:
+    editor: Model.Editor
+    dragManager: R.DragManager
+
+  _onResizeMouseDown: (mouseDownEvent) ->
+    startX = mouseDownEvent.clientX
+    layout = @context.editor.layout
+
+    @context.dragManager.start mouseDownEvent,
+      cursor: "ew-resize"
+      onMove: (moveEvent) =>
+        dx = moveEvent.clientX - startX
+        startX = moveEvent.clientX
+        layout.resizeRightPanel(dx)
+
+  componentDidMount: ->
+    @refs.resize.getDOMNode().addEventListener("mousedown", @_onResizeMouseDown)
+
+  render: ->
+    layout = @context.editor.layout
+
+    R.div { 
+        className: R.cx {
+           RightPanel: true
+           FullScreen: layout.fullScreen
+        }
+        style: {
+          width: layout.rightPanelWidth
+        }
+      },
+      R.div { 
+        className: "RightResize"
+        ref: "resize"
```

Comment:
Instead of giving this a ref in order to make it respond to mousedown (in line 22-23), you can just give this a `onMouseDown` prop. That is, change the line to:

    onMouseDown: @_onResizeMouseDown

This will call @_onResizeMouseDown with the expected `mouseDownEvent`.